### PR TITLE
[ new ] Resolve license issue

### PR DESCRIPTION
**Pull request by L-TChen**
The author field should be fine, I guess.

The `LICENSE` file of Agda does not seem to fit this program, but I am copying it here anyway. I am not confident enough about what LICENSE should actually be. Can anyone make a PR or suggest any change?

### CODE REVIEW

**Comment by L-TChen** [comment]
By the way, this is not necessary if it is going to Hackage. This can just be a submodule of agda.

**Comment by andreasabel** [comment]
Well, for one the text for `Maybe.Strict` doesn't apply here, I suppose.

Couldn't we place this in the public domain, with an "Unlicence"?

**Comment by L-TChen** [comment]
> Well, for one the text for `Maybe.Strict` doesn't apply here, I suppose.

That was my concern about inheriting Agda's LICENSE file, and I did not feel right to use it. 

> Couldn't we place this in the public domain, with an "Unlicence"?

I am very happy with this choice.

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+--------------------------------------------------------------------------------
+
+The file src/full/Agda/Utils/Maybe/Strict.hs (and the following
```
Comment (andreasabel) [inline LICENSE]: The LICENSE file is still here.  Shouldn't it be replaced by a new file reflecting the public domain status?
See the text at https://unlicense.org/

**Review by andreasabel — CHANGES_REQUESTED** [review]
> fix-whitespace was originally written by Nils Anders Danielsson as part of Agda 2 with contributions from Ulf Norell, Andrés Sicard-Ramírez, Andreas Abel, Philipp Hausmann, Jesper Cockx, Vlad Semenov, Liang-Ting Chen, and Ed Nutting.

@nad @UlfNorell @asr @phile314 @jespercockx @semenov-vladyslav @EdNutting : Please waive your copyright or engage in the license discussion if you wish a different license.

**Comment by jespercockx** [comment]
How does one waive a copyright? You may hereby consider mine waived (to whatever small extend I had a claim to it).

File: LICENSE (hunk 1 above)
Comment (L-TChen) [reply LICENSE]: You're right. I overlooked it. `UNLICENSE` is added.

**Comment by phile314** [comment]
I don't think I contributed much if at all; you may do how you please with my contributions. Consider my copyright waived.

**Comment by nad** [comment]
I think the program should have the licence it had when I started implementing it.

**Comment by asr** [comment]
> asr@: Please waive your copyright or engage in the license discussion if you wish a different license.

I waive my copyright.

**Comment by EdNutting** [comment]
Happy to waive my copyright for code I've contributed to this repository.

I assume by "waive" we mean we grant the same license to use any existing contribution we have made to this repository as per the proposed UNLICENSE including all clauses (including but not limited to the 'without-warranty' clause).

[Just being explicit because 'waiving copyright' is not actually a thing - which is why things like the UNLICENSE even exist in the first place]

**Comment by andreasabel** [comment]
@nad wrote:
> I think the program should have the licence it had when I started implementing it.

Then we have to keep it but delete the irrelevant parts.

**Comment by andreasabel** [comment]
Closed in favor of #19.  I salvaged what I could.

**Pull request closed without merging**